peps/pep-0750.rst

```diff
@@ -1,889 +1,1213 @@
+lambda wrapping when lazy evaluation is desired. See the section on
+`Approaches to Lazy Evaluation`_, above, for more information.
+
+
+Making ``Template`` and ``Interpolation`` Into Protocols
+--------------------------------------------------------
 
-        def __len__(self):
-            return 4
+An early version of this PEP proposed that the ``Template`` and ``Interpolation``
+types be runtime checkable protocols rather than concrete types.
 
-        def __iter__(self):
-            return iter((self.getvalue, self.raw, self.conv, self.format_spec))
+In the end, we felt that using concrete types was more straightforward.
 
-However, the additional complexity to support exact round-tripping seems
-unnecessary and is thus rejected.
 
-No Implicit String Concatenation
---------------------------------
+An Additional ``Decoded`` Type
+------------------------------
 
-Implicit tag string concatenation isn't supported, which is `unlike other string literals
+An early version of this PEP proposed an additional type, ``Decoded``, to represent
+the "static string" parts of a template string. This type derived from ``str`` and
+had a single extra ``raw`` attribute that provided the original text of the string.
+We rejected this in favor of the simpler approach of using plain ``str`` and
+allowing combination of ``r`` and ``t`` prefixes.
+
+
+Enable Full Reconstruction of Original Template Literal
+-------------------------------------------------------
+
+Earlier versions of this PEP attempted to make it possible to fully reconstruct
+the text of the original template string from a ``Template`` instance. This was
+rejected as being overly complex.
+
+There are several limitations with respect to round-tripping to the original
+source text:
+
+- ``Interpolation.format_spec`` defaults to ``""`` if not provided. It is therefore
+  impossible to distinguish ``t"{expr}"`` from ``t"{expr:}"``.
+- The debug specifier, ``=``, is treated as a special case. It is therefore not
+  possible to distinguish ``t"{expr=}"`` from ``t"expr={expr}"``.
+- Finally, format specifiers in f-strings allow arbitrary nesting. In this PEP
+  and in the reference implementation, the specifier is eagerly evaluated
+  to set the ``format_spec`` in the ``Interpolation``, thereby losing
+  the original expressions. For example:
+
+.. code-block:: python
+
+    value = 42
+    precision = 2
+    template = t"Value: {value:.{precision}f}"
+    assert template.args[1].format_spec == ".2f"
+
+We do not anticipate that these limitations will be a significant issue in practice.
+Developers who need to obtain the original template string literal can always
+use ``inspect.getsource()`` or similar tools.
+
+
+String Concatenation
+--------------------
+
+Template string concatenation isn't supported, which is `unlike all other string literals
 <https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation>`_.
```

Comment:
```suggestion
Template string concatenation isn't supported, which is :ref:`unlike all other
string literals <python:string-concatenation>`.
```

Reply:
After quite a bit of discussion and agreement amongst all PEP authors, this has been removed from the PEP in favor of _supporting_ concatenation.